src/c/config.c

```diff
@@ -0,0 +1,104 @@
+#include "config.h"
+
+static AppConfig *s_config;
+static Layer *s_layer;
+
+void config_load(AppConfig *config) {
+  // Set default colors
+  config->bg1 = GColorOxfordBlue;
+  config->bg2 = GColorLiberty;
+  config->date1 = GColorWhite;
+  config->date2 = GColorPastelYellow;
+  config->digits = GColorPastelYellow;
+  config->hour = GColorCeleste;
+  config->min = GColorRajah;
+
+  // Load colors from storage if available
+  if (persist_exists(MESSAGE_KEY_BG1)) {
+    config->bg1.argb = persist_read_int(MESSAGE_KEY_BG1);
+  }
+
+  if (persist_exists(MESSAGE_KEY_BG2)) {
+    config->bg2.argb = persist_read_int(MESSAGE_KEY_BG2);
+  }
+
+  if (persist_exists(MESSAGE_KEY_DATE1)) {
+    config->date1.argb = persist_read_int(MESSAGE_KEY_DATE1);
+  }
+
+  if (persist_exists(MESSAGE_KEY_DATE2)) {
+    config->date2.argb = persist_read_int(MESSAGE_KEY_DATE2);
+  }
+
+  if (persist_exists(MESSAGE_KEY_DIGITS)) {
+    config->digits.argb = persist_read_int(MESSAGE_KEY_DIGITS);
+  }
+
+  if (persist_exists(MESSAGE_KEY_HOUR)) {
+    config->hour.argb = persist_read_int(MESSAGE_KEY_HOUR);
+  }
+
+  if (persist_exists(MESSAGE_KEY_MIN)) {
+    config->min.argb = persist_read_int(MESSAGE_KEY_MIN);
+  }
+}
+
+void config_save(AppConfig *config) {
+  persist_write_int(MESSAGE_KEY_BG1, config->bg1.argb);
```

Comment:
I hadn't considered using the MESSAGE_KEYs as persist keys as well. Does this mean that we can never re-order the message keys without breaking backwards compatibility?

If that's true, then it would be a bit safer to define a "SETTINGS_VERSION_KEY" and saving the value 1 in it for now. Then if there are any changes to the way we persist values we could detect which version created the values in the storage.

Similar to this https://github.com/justinjhendrick/pebble-one-handed-minute/blob/41e8bea86906ce41b97132bffab2a4c951374c51/src/c/one-handed-minute.c#L269-L278